feat: Add resolve deployment url (#994)
SAP/ai-sdk-js: .changeset/tiny-trains-tap.md

---
'@sap-ai-sdk/ai-api': minor
---

> [New Functionality] Add `resolveDeploymentUrl()` function to get deployment URL matches the given criteria.

Reviewer: ```suggestion
[New Functionality] Add `resolveDeploymentUrl()` function to dynamically resolve the deployment URL for a given model.
```

> [New Functionality] Add `resolveDeploymentUrl()` function to resolve the deployment URL matches the given criteria.

Reviewer 2: ```suggestion
[New Functionality] Add `resolveDeploymentUrl()` function to resolve the deployment URL that matches the given criteria.
```
